### ISSUE TITLE
hostname property not behaving properly

### ISSUE THREAD

**Issue by Bobo357** [rsyslog/rsyslog#4825]
### Expected behavior
Central rsyslog server saves remote logs to fqdn-ip/year/month/day (eg. client.my.domain.com-10.11.12.23/2022/03/11)
### Actual behavior
Central rsyslog server saves remote logs to hostname-ip/year/month/day (eg. client-10.11.12.23/2022/03/11)
### Steps to reproduce the behavior
It started after upgrading rsyslog clients from 8.2112.0 to 8.2202.0. Configuration was not changed, only upgraded packages.

### Environment
- rsyslog version: 8.2202.0-0adiscon2focal1
- platform: Ubuntu 20.04

Some relevant parts of our rsyslog server configuration:
global(
    preserveFQDN="on")
template(name="RemoteLogSavePath" type="list") {
    constant(value="/srv/syslog/")
    property(name="hostname")
    constant(value="-")
    property(name="fromhost-ip")
    constant(value="/")
    property(name="timegenerated" dateFormat="year")
    constant(value="/")
    property(name="timegenerated" dateFormat="month")
    constant(value="/")
    property(name="timegenerated" dateFormat="day")
    constant(value="/")
    property(name="app-name" )
    constant(value=".log")
ruleset(name="RemoteLogs") {
    $RulesetCreateMainQueue on
    action(type="omfile"
        dynaFileCacheSize="1024"
        dynaFile="RemoteLogSavePath"
        template="RSYSLOG_SyslogProtocol23Format"
        flushOnTXEnd="off"
        asyncWriting="on"
        flushInterval="1"
        ioBufferSize="64k")
}

**Comment by davidelang** [comment]
please log with the template RSYSLOG_DebugFormat and post a sample log so we can 
make sure of what the log contains

David Lang

**Comment by Bobo357** [comment]
So I did a lot of debugging myself and found one significant fact about the issue.
If I restart rsyslog with command `systemctl kill -s HUP rsyslog.service` then RSYSLOG_DebugFormat starts to log with CORRECT fqdn (HOSTNAME: 'host.fqdn'). 
If I restart rsyslog with command `systemctl restart rsyslog` or just reboot the machine then RSYSLOG_DebugFormat shows incomplete hostname (just 'host', without fqdn).
None of the configuration was changed between restarts and I can switch back and forth between different hostname formats with just a simple restart command.
